arraycontext/impl/pytato/__init__.py

```diff
@@ -170,6 +174,22 @@ def transform_loopy_program(self, t_unit):
                          "transform_loopy_program. Sub-classes are supposed "
                          "to implement it.")
 
+    def transform_dag(self, dag: Union["pytato.Array",
+                                       "pytato.DictOfNamedArrays"]
```

Comment:
Would it be possible to standardize this to only `DictOfNamedArrays`?

Reply:
Yep, agreed. Probably easier for the derived classes.